projects/valtimo/zgw/src/lib/modules/documenten-api/components/documenten-api-tags/documenten-api-tags.component.ts

```diff
@@ -0,0 +1,223 @@
+        size: this.pagination.size,
+        search: searchTerm,
+      })
+    ),
+    map((tagPage: Page<DocumentenApiTag>) => {
+      this.pagination = {
+        ...this.pagination,
+        collectionSize: tagPage.totalElements,
+      };
+
+      return tagPage.content;
+    }),
+    tap(tags => {
+      this.loading$.next(false);
+    })
+  );
+
+  public readonly ACTION_ITEMS: ActionItem[] = [
+    {
+      label: 'interface.delete',
+      callback: this.openDeleteModal.bind(this),
+      type: 'danger',
+    },
+  ];
+
+  public pagination: Pagination = {
+    collectionSize: 0,
+    page: 1,
+    size: 10,
+  };
+
+  constructor(
+    private readonly route: ActivatedRoute,
+    private readonly documentenApiTagService: DocumentenApiTagService
+  ) {
+    super();
+  }
+
+  public ngAfterViewInit(): void {
+    this.initFields();
+  }
+
+  public openDeleteModal(tag: DocumentenApiTag): void {
+    this.tagToDelete$.next(tag);
+    this.showDeleteModal$.next(true);
+  }
+
+  public openAddModal(): void {
+    this.addModalClosed$.next(false);
+  }
+
+  public closeModal(closeModalEvent: DocumentenApiColumnModalTypeCloseEvent): void {
+    if (closeModalEvent === 'closeAndRefresh') {
+      this.reload();
+    }
+    this.addModalClosed$.next(true);
+  }
+
+  public showDeleteMultipleModal(): void {
+    this.setSelectedRoleKeys();
+    this.showDeleteMultipleModal$.next(true);
+  }
+
+  private setSelectedRoleKeys(): void {
+    this.selectedRowKeys$.next(
+      this.carbonList.selectedItems.map((tag: DocumentenApiTag) => tag.value)
+    );
+  }
+
+  public paginationClicked(page: number): void {
+    this.pagination = {...this.pagination, page};
+    this.reload();
+  }
+
+  public paginationSet(size: number): void {
+    this.pagination = {...this.pagination, size};
+    this.reload();
+  }
+
+  public confirmDeleteTag(tag: DocumentenApiTag): void {
+    this.documentDefinitionName$
+      .pipe(
+        switchMap(documentDefinitionName =>
+          this.documentenApiTagService.deleteTag(documentDefinitionName, tag.value)
+        )
+      )
+      .subscribe(() => {
+        this.reload();
+      });
+  }
+
+  public confirmDeleteMultipleTag(): void {
+    combineLatest([this.documentDefinitionName$, this.selectedRowKeys$])
+      .pipe(
+        switchMap(([documentDefinitionName, keys]) =>
+          this.documentenApiTagService.deleteTags(documentDefinitionName, keys)
+        )
+      )
+      .subscribe(() => {
+        this.reload();
+      });
+  }
+
+  public searchTermEntered(searchTerm: string): void {
+    this.pagination = {...this.pagination, page: 1};
+    this.searchTerm$.next(searchTerm);
+  }
+
+  private reload(noAnimation = false): void {
+    this._reload$.next(noAnimation ? 'noAnimation' : null);
+  }
+
+  private initFields(): void {
+    this.fields$.next([
+      {
+        key: 'value',
+        label: 'zgw.tags.fields.value',
+        viewType: ViewType.TEXT,
+      },
+    ]);
```

Comment:
You can set the fields directly as a normal property without needing the afterViewInit

Reply:
Changed this